crates/opencascade-sys/tests/triangulation.rs

```diff
@@ -0,0 +1,44 @@
+use opencascade_sys::ffi::{
+    new_point, BRepMesh_IncrementalMesh_ctor, BRepPrimAPI_MakeBox_ctor, BRep_Tool_Triangulation,
+    Poly_Triangulation_Node, TopAbs_ShapeEnum, TopExp_Explorer_ctor, TopLoc_Location_ctor,
+    TopoDS_cast_to_face,
+};
+
+#[test]
+fn it_can_access_mesh_triangulation() {
+    let origin = new_point(0., 0., 0.);
+    let mut cube = BRepPrimAPI_MakeBox_ctor(&origin, 10., 10., 10.);
+
+    let mut mesh = BRepMesh_IncrementalMesh_ctor(cube.pin_mut().Shape(), 0.01);
+
+    let mut triangle_corners = 0;
+
+    let mut edge_explorer =
+        TopExp_Explorer_ctor(mesh.pin_mut().Shape(), TopAbs_ShapeEnum::TopAbs_FACE);
+    while edge_explorer.More() {
+        let face = TopoDS_cast_to_face(edge_explorer.Current());
+        let mut location = TopLoc_Location_ctor();
+
+        let triangulation_handle = BRep_Tool_Triangulation(&face, location.pin_mut());
+        if !triangulation_handle.IsNull() {
+            let triangulation = unsafe { &*triangulation_handle.get() };
```

Comment:
Hmmm, yeah I see what you mean here. I would really like to avoid introducing more `unsafe` functions than is necessary, preferably keeping that layer of code just to `cxx`.

Having particular requirements on calling the function can make it error prone (null check before use, make sure the handle outlives the returned reference, etc.). We should provide an API that doesn't have these requirements, either by writing the more verbose code that uses `cxx`, or providing some abstraction that uses type safety and lifetimes to guarantee it can't be misused.

Reply:
I feel that its pretty normal to have unsafe in a `-sys` crate. Since we are both exposing safe wrappers anyway, I feel its better to do this than to write more glue in C++. This makes cleaner and way more scalable.

Reply:
True, `-sys` crates typically are where the `unsafe` code lives. Currently I'm relying on `cxx` to hide that from the bindings.

I haven't gone through the process of trying to code what you're doing in this PR, so I'll try to go through that and understand the needs here - we might be able to come up with something that is both safe, and convenient to generate and use.

Reply:
So small update. I was thinking about safety here. Really the get() method itself isnt unsafe and we I removed the unsafe marker from cxx. What is unsafe is casting a pointer to a reference without a null check. So using the result of `get` still needs to be in an unsafe block. How about we expose this for now since it reduces glue code and we can hide it if we decided to provide better type wrappers in the `-sys` create.

Reply:
Have you had a chance to look at this?

Reply:
@DSchroer sorry I haven't had time to look closely at this and try to find alternatives. I'll try to find a bit of time today to experiment with this branch but no guarantees.

Reply:
@DSchroer What I'll try to do for now is reduce the overall burden of the C++ wrapper code by implementing your suggestions in the issues you've filed.

I know I was against introducing C++ templates in the past, but maybe there's a clever one we can come up with to deal with opencascade `Handle`s in an ergonomic way, where we can still have the safety guarantees from `cxx` without having to write lots of wrapper code.

Reply:
Ive added a template that allows us to safely bind handles to `xxx -> Result<&T>` functions. Please let me know what you think.

Reply:
@DSchroer looks great! Also really cool that CXX does the translation to a rust function which returns a `Result`, nice.